api/v1beta1/azuremanagedcluster_webhook.go

```diff
@@ -52,18 +52,6 @@ func (r *AzureManagedCluster) ValidateCreate() error {
 			"can be set only if the Cluster API 'MachinePool' feature flag is enabled",
 		)
 	}
-	if r.Spec.ControlPlaneEndpoint.Host != "" {
```

Comment:
if we're removing this I think we should also revert the changes in the controller from https://github.com/kubernetes-sigs/cluster-api-provider-azure/pull/2711/files#diff-98a6a70399877f33a02b37dd04551528506730ad55c62e37980efa5e8100d570R168

Reply:
Honestly it seems like those changes in the controller are academic, as the update webhook will never allow this pathology, right?

So the idea is to just clean up the code and not leave the impression that that if condition will ever be fulfilled?

Reply:
oh I didn't realize we were also enforcing immutability on the update webhook... Perhaps we should just let it be mutable and let the controller overwrite it everytime (basically revert #2711) so we can ensure the endpoint is always set to the correct value. Not the best UX but at least we don't run into weird edge cases. Arguably this is worse UX: `You _may not configure these data when authoring your templates for cluster creation_, otherwise CAPZ will not be able to modify it later with the correct data from the AKS API, and your cluster create operation will fail.`

Reply:
Cool, if we can remove that stern language I think that's a good thing.